hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/table/HoodieTableFactory.java

```diff
@@ -217,31 +217,33 @@ private static void setupHoodieKeyOptions(Configuration conf, CatalogTable table
       }
     }
 
-    // tweak the key gen class if possible
-    final String[] partitions = conf.getString(FlinkOptions.PARTITION_PATH_FIELD).split(",");
-    final String[] pks = conf.getString(FlinkOptions.RECORD_KEY_FIELD).split(",");
-    if (partitions.length == 1) {
-      final String partitionField = partitions[0];
-      if (partitionField.isEmpty()) {
-        conf.setString(FlinkOptions.KEYGEN_CLASS_NAME, NonpartitionedAvroKeyGenerator.class.getName());
-        LOG.info("Table option [{}] is reset to {} because this is a non-partitioned table",
-            FlinkOptions.KEYGEN_CLASS_NAME.key(), NonpartitionedAvroKeyGenerator.class.getName());
-        return;
+    if (StringUtils.isNullOrEmpty(conf.get(FlinkOptions.KEYGEN_CLASS_NAME))) {
+      // tweak the key gen class if possible
```

Comment:
Hi, what kind of keygen clazz do you want to configure for non-partitioned table then ?

Reply:
Hi @danny0405 I want to configure org.apache.hudi.keygen.ComplexAvroKeyGenerator for non partition in hudi-flink side, which not only follow partitions, because in spark side, use complex key as default, but flink can not assign complex for non partition.

Reply:
In https://github.com/apache/hudi/pull/5815, we have fixed the spark sql to use `NonpartitionedKeyGenerator` for non partitioned table.

Reply:
But user cannot assign keygen_class seems not friendly.

Reply:
Generally that's true, but non-partitioned table is a special case and hudi configure the keygen clazz transparently for user.

Reply:
Hudi configure keygen clazz auto is great, so the option should not exists, once configured but not effect, is it strange？The code in spark has changed to follow hudi-partition way, but in historical data, if the layout of non-partitioned table with complex key by spark, the only chance for hudi-flink it to configure keygen.

Reply:
Hudi configure keygen clazz auto is great, so the option should not exists, once configured but not effect, is it strange？The code in spark has changed to follow hudi-partition way, but in historical data, if the layout of non-partitioned table with complex key by spark, the only chance for hudi-flink it to configure keygen.

Reply:
That's true, but we better use the right key gen clazz for better performance

Reply:
Of course, shorter key will gain better performance, but this option should also take effect, right？